table/block_based/block_based_table_reader_sync_and_async.h

```diff
@@ -304,6 +308,17 @@ DEFINE_SYNC_AND_ASYNC(void, BlockBasedTable::RetrieveMultipleBlocks)
     }
     statuses[idx_in_batch] = s;
   }
+
+  if (use_fs_scratch) {
+    // Free the allocated scratch buffer by fs here as read requests might have
+    // been combined into one.
+    for (FSReadRequest& req : read_reqs) {
```

Comment:
The fs_scratch buffers will get freed when they go out of scope anyway. But no harm in freeing here to be safe.